File: SoloLearn/Code Coach/Snowballing Numbers/SnowballingNumbers.java
import java.util.Scanner;

public class SnowballingNumbers {

    public static String snowball(Scanner scanner) {
        int n = scanner.nextInt(), val, total = 0;
        for (int i=0; i<n; i++) {
            val = scanner.nextInt();
            if (val <= total)
                return "false";
            total += val;
        }
        return "true";
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println(snowball(input));
        input.close();
    }
}

File: Code Coach/Snowballing Numbers/SnowballingNumbers.c
#include <stdio.h>

const char* snowball(int n) {
    int val, total = 0;
    for (int i=0; i<n; i++) {
        scanf("%d", &val);
        if (val <= total)
            return "false";
        total += val;
    }
    return "true";
}

int main() {
    int n;
    scanf("%d", &n);
    printf("%s", snowball(n));
    return 0;
}